### ISSUE TITLE
Can the output of `brew bundle dump` be sorted consistently?

### ISSUE THREAD

**Issue by paddyroddy** [Homebrew/homebrew-bundle#1263]
I'm tracking my `.Brewfile` in `git` and I see occasionally packages move around. I'm not sure how things are sorted currently, but could it be done on package name?

I'm running the command
```sh
brew bundle dump --describe --force --file $HOME/.Brewfile
```

**Comment by MikeMcQuaid** [comment]
> I'm not sure how things are sorted currently

It's a topological sort. The ordering is important as otherwise you'll install dependencies in the not the order specified in the `Brewfile`.



> but could it be done on package name?

It could but, due to the above, it should not be.

**Comment by paddyroddy** [comment]
> It's a topological sort. The ordering is important as otherwise you'll install dependencies in the not the order specified in the `Brewfile`.

Fair enough. Although, I _think_ that as long as `tap` remains at the top then it's fine. FWIW I have done the following"

```bash
#!/usr/bin/env bash
BREW_FILE=$HOME/.Brewfile
BREW_TYPES=(
    tap
    brew
    cask
    mas
)
TMP_FILE=/tmp/.Brewfile_unsorted

# update homebrew list
brew bundle dump --force --file $TMP_FILE

# sort the final output
## clear current contents
> $BREW_FILE
for type in ${BREW_TYPES[@]}; do
    ## extract lines starting with "type" and sort them
    grep --dereference-recursive "^${type}" $TMP_FILE | sort >> $BREW_FILE
done
## remove temporary file
rm $TMP_FILE
```

**Comment by MikeMcQuaid** [comment]
> then it's fine

It's fine if you're happy for dependencies to be installed not when the `Brewfile` things they will be but before that and the error messages to not be clear about what package has failed to install.
